### PR TITLE
Test geojson to value mapping

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes

Added direct mapping for geojson -> Value mapping to save some cpu cycles. 

In general seems faster 2 times and more, depending on the size of the geojson.
However, `Expression.fromRaw` seem compatibly fast, so the question is - do we want to introduce these optimizations or just add some words suggesting using `fromRaw`, maybe on background thread for large expressions @mapbox/maps-android ? 

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt [hunk 1]
```diff
@@ -40,20 +43,59 @@ class DistanceExpressionActivity : AppCompatActivity() {
...
+          fillLayer(CIRCLE_ID2, CIRCLE_ID2) {
+            fillOpacity(0.5)
+            fillColor("#00b2d0")
+          },
+          below = POI_LABEL
+        )
       }
     ) {
-      val symbolLayer = it.getLayer("poi-label") as SymbolLayer
+      val symbolLayer = it.getLayer(POI_LABEL) as SymbolLayer
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:68]: For sake of testing

File: extension-style/src/main/java/com/mapbox/maps/extension/style/expressions/generated/Expression.kt [hunk 2]
```diff
@@ -2301,6 +2305,139 @@ class Expression : Value {
     fun cos(block: ExpressionBuilder.() -> Unit): Expression =
       ExpressionBuilder("cos").apply(block).build()
 
+    fun BoundingBox.toValue(): Value {
```
Comment (yunikkk) [inline extension-style/src/main/java/com/mapbox/maps/extension/style/expressions/generated/Expression.kt:2308]: Previously geojson was [serialized](https://github.com/mapbox/mapbox-java/blob/a86a6148b87ff3e4a9bd1096eaed5715b80db5e1/services-geojson/src/main/java/com/mapbox/geojson/BaseGeometryTypeAdapter.java) to json, then parsed back from string.

File: app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt [hunk 3]
```diff
@@ -40,20 +43,59 @@ class DistanceExpressionActivity : AppCompatActivity() {
...
+      for (i in 0..iter) {
+        Expression.distance(lineString)
+      }
+      println("Distance time : ${System.nanoTime() - startNano}")
+
+      startNano = System.nanoTime()
+      for (i in 0..iter) {
+        Expression.fromRaw(LINESTRING_JSON)
+      }
+      println("From raw : ${System.nanoTime() - startNano}")
```
Comment (jush) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:91]: Nit: ICYMI you can use Kotlin [measureNanoTime](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/measure-nano-time.html)
```suggestion
val timeInNanos = measureNanoTime {
      repeat(iter) {
        Expression.fromRaw(LINESTRING_JSON)
      }
}
println("From raw : ${timeInNanos}")
```

**Comment by pengdev** [comment]
> In general seems faster 2 times and more, depending on the size of the geojson.
> However, Expression.fromRaw seem compatibly fast, so the question is - do we want to introduce these optimizations or just add some words suggesting using fromRaw, maybe on background thread for large expressions

@yunikkk I think user might prefer to use strongly typed GeoJson class, would it be possible to move this Utility to Value class itself? so it can be reused in other places as well? e.g. `Value.fromGeometry()`,`Value.fromFeature()`,`Value.fromFeatureCollection()`?
And inside the distance(geometry), we just use the fast conversion under the hood?

**Comment by yunikkk** [comment]
Closing as discussed in favor of using `Expression.fromRaw` on the client end on the BG thread

**Pull request closed without merging**